build_config/lib/src/build_config.dart

```diff
@@ -145,24 +145,27 @@ Map<String, T> _normalizeBuilderDefinitions<T>(
         normalizeBuilderKeyDefinition(key, packageName), definition));
 
 String _prettyPrintCheckedFromJsonException(CheckedFromJsonException err) {
-  var yamlMap = err.map as YamlMap;
-
-  var yamlKey = yamlMap.nodes.keys.singleWhere(
-      (k) => (k as YamlScalar).value == err.key,
-      orElse: () => null) as YamlScalar;
-
   var message = 'Could not create `${err.className}`.';
-  if (yamlKey == null) {
-    assert(err.key == null);
-    message = '${yamlMap.span.message(message)} ${err.innerError}';
+  if (err.key == null) {
+    if (err.map is YamlMap) {
```

Comment:
nit: if you pull out the `map` as a local var and do the type check there, you shouldn't have to cast it below.

Reply:
Done